engine/components-api/src/main/java/com/cloud/agent/AgentManager.java

```diff
@@ -50,6 +50,10 @@ public interface AgentManager {
     ConfigKey<Integer> ReadyCommandWait = new ConfigKey<Integer>("Advanced", Integer.class, "ready.command.wait",
             "60", "Time in seconds to wait for Ready command to return", true);
 
+    ConfigKey<String> GranularWaitTimeForCommands = new ConfigKey<>("Advanced", String.class, "commands.timeout", "",
+            "This timeout overrides the wait global config. This holds a comma separated key value pairs containing timeout (in seconds) for specific commands. " +
+                    "For example: DhcpEntryCommand=600, SavePasswordCommand=300, VmDataCommand=300", true);
```

Comment:
I would not make this one dynamic as the whole array needs to be loaded on every command sent to every agent. this will strain the system. The array should be constructed on load and be in memory readily available from then on. I.E.
```suggestion
                    "For example: DhcpEntryCommand=600, SavePasswordCommand=300, VmDataCommand=300", false);
```

Reply:
Agreed @DaanHoogland, modified the code accordingly